mage_ai/io/qdrant.py

```diff
@@ -0,0 +1,136 @@
+from typing import List
+
+from pandas import DataFrame
+from qdrant_client import QdrantClient
+from qdrant_client.http import models
+
+from mage_ai.io.base import BaseIO
+from mage_ai.io.config import BaseConfigLoader, ConfigKey
+
+
+class Qdrant(BaseIO):
+    def __init__(
+            self,
+            collection: str,
+            path: str = None,
+            verbose: bool = True,
+            **kwargs,) -> None:
+        """
+        Initializes connection to qdrant db.
+        """
+        super().__init__(verbose=verbose)
+        self.collection = collection
+        if path is None:
+            self.client = QdrantClient(':memory:')
+        else:
+            self.client = QdrantClient(path=path)
+
+    @classmethod
+    def with_config(cls, config: BaseConfigLoader) -> 'Qdrant':
+        return cls(
+            collection=config[ConfigKey.QDRANT_COLLECTION],
+            path=config[ConfigKey.QDRANT_PATH],
+        )
+
+    def create_collection(
+            self,
+            vector_size: int,
+            distance: models.Distance = None,
+            collection_name: str = None):
+        """
+            Create collection in qdrant db.
+            Args:
+                vector_size (int): dimension size of the vector.
+                distance (models.Distance): distance metric to use.
+                collection_name (str): name of the collection.
+                Defaults to the name defined in io_config.yaml.
+            Returns:
+                collection created.
+        """
+        collection_name = self.collection if collection_name is None else collection_name
+        distance = models.Distance.COSINE if distance is None else distance
+        return self.client.create_collection(
+            collection_name=collection_name,
+            vectors_config=models.VectorParams(
+                size=vector_size,
+                distance=distance),
+        )
+
+    def load(
+        self,
+        limit_results: int,
+        query_vector: List,
+        collection_name: str = None,
+        **kwargs,
+    ) -> DataFrame:
+        """
+            Loads the data from Qdrant with query_vector.
+            Args:
+                limit_results (int): Number of results to return.
+                query_vector (List): vector list used to query.
+                collection_name (str): name of the collection.
+                Defaults to the name defined in io_config.yaml.
+            Returns:
+                DataFrame: Data frame object loaded with data from qdrant
+        """
+        # Assume collection is already created and exists.
+        collection_name = self.collection if collection_name is None else collection_name
```

Comment:
this can be simplified to `collection_name = collection_name or self.collection`